### ISSUE TITLE
[DX] Form API: #title_display has no effect on fieldsets

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4616]
This is a known issue that we've inherited from D7, which our Drupal brethren have categorized as a documentation task and seem to not be willing to do something about: https://www.drupal.org/project/documentation/issues/2055345

> Is this a doc update or a defect? I'd really like to hide the fieldset title using this functionality and was pretty excited that I could only find out that it doesn't actually work.

> If you want to hide the title, use a preprocess function and remove #title.
>
> The rest of the options in #title_display do not seem relevant to fieldsets... and at this point in D7, there is nearly zero chance that the code would be changed.

We should fix this.

**Comment by BWPanda** [comment]
@klonos I know you link to the relevant D7 issue, but are you also able to put a summary of the problem as the first thing in the OP? That way, when the title's a bit ambiguous, we know exactly what the issue is about.

Many time I've seen an issue and still not know what it was about after reading the OP.

**Comment by klonos** [comment]
You are absolutely right @BWPanda, I've updated the issue summary (hope I've done a good job at explaining the problem).

**Comment by klonos** [comment]
There you go: https://github.com/backdrop/backdrop/pull/3304 ...I've never worked on FAPI elements before, but the PR works as expected when I tested it with #4615 on my local. Here's the code changes from that form:

```php
  $form['syslog_entries'] = array(
    '#type' => 'fieldset',
    '#title' => t('Syslog entries'),
    '#title_display' => 'invisible',
    '#collapsible' => FALSE,
  );
  $form['syslog_entries']['syslog_format'] = array(
    '#type' => 'textarea',
    '#title' => t('Syslog entry format'),
    '#default_value' => $config->get('log_format'),
    '#description' => t('For supported variables, refer to the "Available varaibles" fieldset below.'),
  );
```

Before:

![image](https://user-images.githubusercontent.com/2423362/92995602-95832000-f50d-11ea-8008-66cf7ed6cd4b.png)

After:

![image](https://user-images.githubusercontent.com/2423362/92995588-82705000-f50d-11ea-9031-5bfc86259128.png)

**Comment by indigoxela** [comment]
What happens if a fieldset is set to "collapsible" and the title is hidden?

**Comment by klonos** [comment]
> What happens if a fieldset is set to "collapsible" and the title is hidden?

The `'#collapsible'` property depends on the presence of the fieldset title/legend, so when the legend is missing, the fieldset is _always_ rendered as expanded and non-collapsible. I believe that this is expected behavior, similar to the fact that the only supported value for `'#title_display'` for fieldsets should be `'invisible'` (normally, there's also `'before'`, `'after'`, and `'identity'` - see https://github.com/backdrop/backdrop/blob/1.x/core/includes/form.inc#L4687).

**Comment by indigoxela** [comment]
> The '#collapsible' property depends on the presence of the fieldset title/legend, so when the legend is missing, the fieldset is always rendered as expanded and non-collapsible.

If I get this right, you suggest to make the '#title_display' more consistent and at the same time make '#collapsible' less consistent (silently ignored). Or am I missing something?

**Comment by kiamlaluno** [comment]
The code handling `#title_display` should also correctly handle the fieldsets with a `#group` value because fieldsets used in vertical tabs need a title.
